# **Sports Analytics**

## **Using F I F A 2 3 Official Dataset Predict Player Performance**

#### **The dataset contains 63 columns, each representing a different attribute of a football player.**

ID: A unique identifier for each player.

Name: The player's full name.

Age: The player's age.

Photo: A URL to the player's photo.

Nationality: The player's nationality.

Flag: A URL to the flag of the player's country.

Overall: The player's overall rating based on various attributes.

Potential: The player's potential rating, which indicates their peak performance.

Club: The name of the club the player is currently playing for.

Club Logo: A URL to the logo of the player's club.

Value: The player's estimated market value.

Wage: The player's weekly wage.

Special: A special rating based on how unique or outstanding the player's skills are.

Preferred Foot: Whether the player prefers their left or right foot.

International Reputation: A score indicating the player's international reputation (higher means more well-known globally).

Weak Foot: The player's skill level with their non-dominant foot.

Skill Moves: The player's proficiency with skill moves (rated out of 5 stars).

Work Rate: The player's work rate in both attack and defense (e.g., "High/Medium").

Body Type: The player's physical build (e.g., lean, stocky).

Real Face: Whether the player's face is realistically rendered in the game (True/False).

Position: The player's primary playing position (e.g., ST, CM).

Jersey Number: The player's squad number.

Joined: The date the player joined their current club.

Loaned From: The club the player is loaned from (if applicable).

Contract Valid Until: The year the player's contract with their current club expires.

Height: The player's height (in feet and inches).

Weight: The player's weight (in pounds).

Crossing: The player's ability to deliver accurate crosses.

Finishing: The player's ability to score from chances.

HeadingAccuracy: The player's accuracy with headers.

ShortPassing: The player's ability to pass the ball accurately over short distances.

Volleys: The player's ability to strike the ball mid-air (volleys).

Dribbling: The player's ability to move with the ball while controlling it closely.

Curve: The player's ability to bend the ball in the air (curve shots or passes).

FKAccuracy: The player's ability to take accurate free kicks.

LongPassing: The player's ability to deliver long-distance passes accurately.

BallControl: The player's ability to control the ball in various situations.

Acceleration: How quickly the player can reach top speed.

SprintSpeed: The player's top running speed.

Agility: How quickly the player can change direction.

Reactions: The player's reaction time to in-game events.

Balance: The player's ability to maintain balance, especially in duels or when dribbling.

ShotPower: The power behind the player's shots.

Jumping: The player's jumping ability.

Stamina: The player's endurance, representing how long they can maintain their energy.

Strength: The player's physical strength in duels.

LongShots: The player's accuracy when shooting from long distances.

Aggression: The player's intensity in challenging for the ball.

Interceptions: The player's ability to read the game and intercept passes.

Positioning: The player's ability to position themselves well on the field.

Vision: The player's awareness of their teammates' positions.

Penalties: The player's accuracy in taking penalty kicks.

Composure: The player's ability to remain calm under pressure.

Marking: The player's ability to mark opponents effectively.

StandingTackle: The player's skill at standing tackles.

SlidingTackle: The player's skill at sliding tackles.

GKDiving: (Goalkeepers only) The goalkeeper's diving ability.

GKHandling: (Goalkeepers only) The goalkeeper's ability to catch the ball.

GKKicking: (Goalkeepers only) The goalkeeper's kicking accuracy and distance.

GKPositioning: (Goalkeepers only) The goalkeeper's positioning awareness.

GKReflexes: (Goalkeepers only) The goalkeeper's reaction speed.

Best Position: The player's best-suited position on the field.

Best Overall Rating:


In [167]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [168]:
df=pd.read_csv("C:\\Users\\NAYANA\\Downloads\\FIFA17_official_data.csv~\\FIFA17_official_data.csv")

## **Data Understanding**

In [170]:
df.head()

ID            Name  Age  \
0  176580       L. Suárez   29   
1  178518   R. Nainggolan   28   
2  181872        A. Vidal   29   
3  197445        D. Alaba   24   
4  195864        P. Pogba   23   

                                              Photo Nationality  \
0  https://cdn.sofifa.com/players/176/580/17_60.png     Uruguay   
1  https://cdn.sofifa.com/players/178/518/17_60.png     Belgium   
2  https://cdn.sofifa.com/players/181/872/17_60.png       Chile   
3  https://cdn.sofifa.com/players/197/445/17_60.png     Austria   
4  https://cdn.sofifa.com/players/195/864/17_60.png      France   

                                  Flag  Overall  Potential               Club  \
0  https://cdn.sofifa.com/flags/uy.png       92         92       FC Barcelona   
1  https://cdn.sofifa.com/flags/be.png       86         86               Roma   
2  https://cdn.sofifa.com/flags/cl.png       87         87  FC Bayern München   
3  https://cdn.sofifa.com/flags/at.png       86         89  FC Bayern München   
4  https://cdn.sofifa.com/flags/fr.png       88         94  Manchester United   

                                       Club Logo  ... Marking StandingTackle  \
0  https://cdn.sofifa.com/teams/241/light_30.png  ...    30.0           45.0   
1   https://cdn.sofifa.com/teams/52/light_30.png  ...    78.0           85.0   
2   https://cdn.sofifa.com/teams/21/light_30.png  ...    76.0           89.0   
3   https://cdn.sofifa.com/teams/21/light_30.png  ...    81.0           83.0   
4   https://cdn.sofifa.com/teams/11/light_30.png  ...    68.0           73.0   

   SlidingTackle GKDiving  GKHandling  GKKicking  GKPositioning GKReflexes  \
0           38.0     27.0        25.0       31.0           33.0       37.0   
1           88.0     11.0        11.0       14.0            8.0       11.0   
2           84.0      4.0         2.0        4.0            2.0        4.0   
3           83.0      5.0         7.0       14.0           15.0        9.0   
4           73.0      5.0         6.0        2.0            4.0        3.0   

  Best Position Best Overall Rating  
0            ST                88.0  
1           CDM                84.0  
2           CDM                85.0  
3            LB                84.0  
4           CAM                85.0  

[5 rows x 63 columns]

In [171]:
df.shape

(17560, 63)

The dataset has 17,560 rows and 63 columns. This means it contains data for 17,560 players, with 63 different features for each player.

In [173]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 17560 entries, 0 to 17559
Data columns (total 63 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   ID                        17560 non-null  int64  
 1   Name                      17560 non-null  object 
 2   Age                       17560 non-null  int64  
 3   Photo                     17560 non-null  object 
 4   Nationality               17560 non-null  object 
 5   Flag                      17560 non-null  object 
 6   Overall                   17560 non-null  int64  
 7   Potential                 17560 non-null  int64  
 8   Club                      17317 non-null  object 
 9   Club Logo                 17560 non-null  object 
 10  Value                     17560 non-null  object 
 11  Wage                      17560 non-null  object 
 12  Special                   17560 non-null  int64  
 13  Preferred Foot            17560 non-null  object 
 14  Intern

**Data Types:**

i.   5 integer columns (int64)

ii.  39 float columns (float64)

iii. 19 object columns (object)

**Missing Values**

i.   Club: Missing in 243 rows.

ii.  Joined: Missing in 1,695 rows.

iii. Loaned From

In [209]:
# drop column that have any importance in player performance
df.drop(columns=['ID', 'Photo', 'Flag', 'Club Logo'], axis=1, inplace=True)

## **Summary Statistics**

In [212]:
df.describe().T

count         mean         std    min     25%  \
Age                       17560.0    24.576310    4.653748   15.0    21.0   
Overall                   17560.0    66.155125    7.013210   45.0    62.0   
Potential                 17560.0    70.938326    6.140487   45.0    67.0   
Special                   17560.0  1595.468223  269.862012  751.0  1452.0   
International Reputation  17560.0     1.111902    0.385032    1.0     1.0   
Weak Foot                 17560.0     2.935820    0.657838    1.0     3.0   
Skill Moves               17560.0     2.305410    0.746962    1.0     2.0   
Jersey Number             17541.0    20.411265   17.047842    1.0     9.0   
Crossing                  17560.0    49.778360   18.423713    6.0    38.0   
Finishing                 17560.0    45.209909   19.366051    2.0    29.0   
HeadingAccuracy           17560.0    52.382973   17.445995    4.0    45.0   
ShortPassing              17560.0    58.204442   14.874607   10.0    53.0   
Volleys                   17560.0    43.220444   17.684370    3.0    30.0   
Dribbling                 17560.0    54.891401   18.872939    4.0    47.0   
Curve                     17560.0    47.249374   18.415806    6.0    34.0   
FKAccuracy                17560.0    43.374886   17.657061    4.0    31.0   
LongPassing               17560.0    52.449658   15.501135    9.0    42.0   
BallControl               17560.0    58.027733   16.763558    5.0    53.0   
Acceleration              17560.0    65.148918   14.504689   11.0    57.0   
SprintSpeed               17560.0    65.383998   14.189398   11.0    58.0   
Agility                   17560.0    63.249943   14.607362   11.0    55.0   
Reactions                 17560.0    61.730866    9.225748   30.0    55.0   
Balance                   17560.0    63.990604   13.790504   10.0    56.0   
ShotPower                 17560.0    55.626879   17.523230    3.0    46.0   
Jumping                   17560.0    64.934738   11.489039   15.0    58.0   
Stamina                   17560.0    63.392654   15.563321   10.0    57.0   
Strength                  17560.0    65.075911   12.525802   21.0    57.0   
LongShots                 17560.0    47.394647   19.154503    4.0    33.0   
Aggression                17560.0    55.954328   17.435284    7.0    44.0   
Interceptions             17560.0    46.748804   20.513954    3.0    26.0   
Positioning               17560.0    49.632005   19.387491    2.0    38.0   
Vision                    17560.0    52.805866   14.472393   10.0    43.0   
Penalties                 17560.0    49.108371   15.824783    7.0    39.0   
Composure                 17560.0    55.979954   13.301553    5.0    47.0   
Marking                   17560.0    44.225399   21.532570    4.0    22.0   
StandingTackle            17560.0    47.460194   21.785704    3.0    26.0   
SlidingTackle             17560.0    45.540888   21.459058    5.0    24.0   
GKDiving                  17560.0    16.811333   17.796100    1.0     8.0   
GKHandling                17560.0    16.536731   16.953668    1.0     8.0   
GKKicking                 17560.0    16.441856   16.589324    1.0     8.0   
GKPositioning             17560.0    16.596185   17.137032    1.0     8.0   
GKReflexes                17560.0    16.887927   18.034308    1.0     8.0   
Best Overall Rating       17560.0    65.620615    6.940057   44.0    61.0   

                             50%     75%     max  
Age                         24.0    28.0    46.0  
Overall                     66.0    71.0    94.0  
Potential                   71.0    75.0    94.0  
Special                   1633.0  1786.0  2291.0  
International Reputation     1.0     1.0     5.0  
Weak Foot                    3.0     3.0     5.0  
Skill Moves                  2.0     3.0     5.0  
Jersey Number               18.0    27.0    99.0  
Crossing                    54.0    64.0    91.0  
Finishing                   48.0    61.0    95.0  
HeadingAccuracy             56.0    65.0    94.0  
ShortPassing               

## Handling Missing Values

In [215]:
missing_values=df.isna().sum() # check the missing value in the dataset

In [217]:
missing_values_col=missing_values[missing_values > 0]
missing_values_col

Club                      243
Position                   19
Jersey Number              19
Joined                   1695
Loaned From             16227
Contract Valid Until      362
dtype: int64

i. **Club** column has **243** missing values, Club is used as an identifier, so we cant replace the missing value with median or mode. **drop** the rows having missing value is a good decision.
ii. **Position** column has **19** missing value, we can replace the value with mode.
iii. **Jersey Number** column has 19 missing values
iv.
ii. **Loaned From** column has **16227** missing values out of 17560. So we have to **drop** it.

In [ ]:
# check 

### EDA

In [182]:
#df.drop(columns=['ID', 'Photo', 'Flag', 'Overall', 'Potential', 'Club Logo',
        # 'Value', 'Wage', 'Special', 'Preferred Foot', 'Weak Foot', 'Work Rate', 'Body Type', 'Real Face', 
        # 'Jersey Number', 'Contract Valid Until',
        # 'GKDiving', 'GKHandling',
        # 'GKKicking', 'GKPositioning', 'GKReflexes'], axis=1, inplace=True)